src/charm.py

```diff
@@ -1,117 +1,278 @@
+        """
+        return self._tls.get_tls_relation() is not None
+
+    def _reconcile(self) -> None:
+        """Reconcile charm status based on configuration and integrations."""
+        tls_certificates_relation = self._tls.get_tls_relation()
+        if not tls_certificates_relation:
+            self.unit.status = BlockedStatus("Waiting for TLS.")
+            return
+        self.unit.status = ActiveStatus()
+
+    def _on_config_changed(self, _: Any) -> None:
+        """Handle the config-changed event."""
+        self._reconcile()
+
+    def _on_start(self, _: Any) -> None:
+        """Handle the start event."""
+        self._reconcile()
+
+    def _on_get_certificate_action(self, event: ActionEvent) -> None:
+        """Triggered when users run the `get-certificate` Juju action.
+
+        Args:
+            event: Juju event
+        """
+        hostname = event.params["hostname"]
+        tls_certificates_relation = self._tls.get_tls_relation()
+        if not tls_certificates_relation:
+            event.fail("Certificates relation not created.")
+            return
 
-    def _on_httpbin_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
-        """Define and start a workload using the Pebble API.
+        tls_rel_data = tls_certificates_relation.data[self.app]
+        if not tls_rel_data.get(f"certificate-{hostname}"):
+            event.fail("Certificate not available")
+            return
 
-        Change this example to suit your needs. You'll need to specify the right entrypoint and
-        environment configuration for your specific workload.
+        event.set_results(
+            {
+                f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
+                f"ca-{hostname}": tls_rel_data[f"ca-{hostname}"],
+                f"chain-{hostname}": tls_rel_data[f"chain-{hostname}"],
+            }
+        )
 
-        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
+    def _on_certificates_relation_created(self, event: RelationCreatedEvent) -> None:
+        """Handle the TLS Certificate relation created event.
 
         Args:
-            event: event triggering the handler.
+            event: The event that fires this method.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", self._pebble_layer, combine=True)
-        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
-        container.replan()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
-        self.unit.status = ops.ActiveStatus()
+        if not self._are_relations_ready():
+            self.unit.status = WaitingStatus("Waiting for certificates relation to be created")
```

Comment:
Since this is running in the `_on_certificates_relation_created` hook I think we can be sure that the relation will be ready at some point, so nothing for the user to do here. We do block the charm here: https://github.com/canonical/gateway-api-integrator-operator/blob/a40b631991d954e5afdf5a349766e0677ec1f716/src/charm.py#L101-L102 since at this point it's the operator's action to create a relation.